fsspec/implementations/http.py

```diff
@@ -147,6 +147,17 @@ def size(self, url):
         """Size in bytes of the file at path"""
         return file_size(url, session=self.session, **self.kwargs)
 
+    def info(self, url, size_policy='head'):
+        try:
+            size = file_size(url, self.session, size_policy)
+            return {'name': url, 'size': size, 'type': 'file'}
+        except:
+            try:
+                self.session.get(url, stream=True)
```

Comment:
Any reason not to try `file_size(url, self.session, 'get')` if `head` fails, and then fallback to `None`? The `get` should also be a good backup check for file existence, with the benefit you may be able to get size from it?

I *think* the following may work:

```python
def info(self, url):
    for policy in ['head', 'get']:
        try:
            size = file_size(url, self.session, policy)
            break
        except ValueError as exc:
            size = None  # the ValueError from in `file_size`, would maybe be better just return `None` in `file_size`
            break
        except Exception:
            pass
    else:
        raise FileNotFoundError(url)
    return {"name": url, "size": size, "type": "file"}
```

Reply:
OK, I included it. Good to have extra eyes on this code...
The only potential problem with this is making extra requests for every file, but that should be better than exceptions that are hard to understand or remedy.